catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/CswUnmarshallHelper.java

```diff
@@ -132,179 +128,109 @@ public static void removeExistingAttributes(Map<String, String> cswAttrMap,
+            HierarchicalStreamReader hreader, CswAxisOrder cswAxisOrder,
+            Map<String, String> namespaceMap) {
 
         StringWriter metadataWriter = new StringWriter();
         HierarchicalStreamReader reader = XStreamAttributeCopier.copyXml(hreader,
                 metadataWriter,
                 namespaceMap);
 
-        MetacardImpl mc = new MetacardImpl(CSW_METACARD_TYPE);
-        Map<String, Attribute> attributes = new TreeMap<>();
+        MetacardImpl mc = new MetacardImpl(metacardType);
 
         while (reader.hasMoreChildren()) {
             reader.moveDown();
 
             String nodeName = reader.getNodeName();
             LOGGER.debug("node name: {}.", nodeName);
-
             String name = getCswAttributeFromAttributeName(nodeName);
-
             LOGGER.debug("Processing node {}", name);
-            AttributeDescriptor attributeDescriptor =
-                    CSW_METACARD_TYPE.getAttributeDescriptor(name);
-
-            Serializable value = null;
-
-            // If XML node name matched an attribute descriptor in the
-            // metacardType AND
-            // the XML node has a non-blank value OR this is geometry/spatial
-            // data,
-            // then convert the CSW Record's property value for this XML node to
-            // the
-            // corresponding metacard attribute's value
-            if (attributeDescriptor != null && (StringUtils.isNotBlank(reader.getValue())
-                    || BasicTypes.GEO_TYPE.equals(attributeDescriptor.getType()))) {
-                value = convertRecordPropertyToMetacardAttribute(attributeDescriptor.getType()
-                        .getAttributeFormat(), reader, cswAxisOrder);
-            }
 
-            if (null != value) {
-                if (attributeDescriptor.isMultiValued()) {
-                    if (attributes.containsKey(name)) {
-                        AttributeImpl attribute = (AttributeImpl) attributes.get(name);
-                        attribute.addValue(value);
+            if (DefaultCswRecordMap.hasDefaultMetacardFieldFor(name)) {
+                String test = DefaultCswRecordMap.getDefaultMetacardFieldFor(name);
+
+                AttributeDescriptor attributeDescriptor = metacardType.getAttributeDescriptor(test);
+                Serializable value =
+                        convertRecordPropertyToMetacardAttribute(attributeDescriptor.getType()
+                                .getAttributeFormat(), reader, cswAxisOrder);
+
+                if (isNotEmpty(value)) {
+
+                    String attributeName = attributeDescriptor.getName();
+
+                    if (attributeName.equals(Core.THUMBNAIL)) {
+                        String uri = new String((byte[]) value, StandardCharsets.UTF_8);
+                        URL url;
+                        InputStream is = null;
+                        try {
+                            url = new URL(uri);
+                            is = url.openStream();
```

Comment:
@stustison - the thumbnail isn't included in the Csw response and that is why we open a connection to go get it.  You definitely make a valid point about checking the response.  As a note, this wasn't introduced in this PR.  In fact, the CSW code has been doing this for a long time.  I'd suggest we open a separate ticket for this issue.